src/actions/classifier.js

```diff
@@ -35,3 +51,143 @@ export function fetchWorkflow(workflowID) {
    })
   }
 }
+
+export function fetchTutorials(workflowID) {
+  return dispatch => {
+    return new Promise ((resolve) => {
+      apiClient.type('tutorials').get({workflow_id: workflowID}).then(([tutorial]) => {
+        const tutorialResource = tutorial
+        dispatch(setState(`classifier.tutorial.${workflowID}`, tutorialResource))
+        if (!isEmpty(tutorial)) {
+          let mediaByID = {}
+          tutorialResource.get('attached_images').then((mediaResources) => {
+            forEach((mediaResource) => mediaByID[mediaResource.id] = mediaResource, mediaResources)
+            dispatch(setState(`classifier.tutorial.${workflowID}.mediaResources`, mediaByID))
+            return resolve()
+          }).catch(() => {
+            return resolve()
+          })
+        } else {
+          return resolve()
+        }
+      }).catch(() => { //does not exist for this project, that is OK
+        dispatch(setState(`classifier.tutorial.${workflowID}`, {}))
+        return resolve()
+      })
+    })
+  }
+}
+
+export function fetchProject(workflowID) {
+  return (dispatch, getState) => {
+    const workflow = getState().classifier.workflow[workflowID]
+    const projectID = workflow.links.project
+    return new Promise ((resolve, reject) => {
+      apiClient.type('projects').get({id: projectID}).then(([project]) => {
+        dispatch(setState(`classifier.project.${workflowID}`, project))
+        return resolve()
+      }).catch(() => {
+        return reject()
+      })
+   })
+  }
+}
+
+export function setupProjectPreferences(workflowID) {
+  return (dispatch, getState) => {
+    const workflow = getState().classifier.workflow[workflowID]
+    const projectID = workflow.links.project
+
+    return new Promise ((resolve, reject) => {
+      if (getState().user.isGuestUser){
+        return resolve()
+      }
+
+      dispatch(getAuthUser()).then((userResource) => {
+        return userResource
```

Comment:
Am I missing something here? It seems a bit odd to have a repeating `.then(userResource)`.